tests/consensus_tests/test_shard_wal_delta_transfer.py

```diff
@@ -409,9 +391,11 @@ def test_shard_wal_delta_transfer_fallback(capfd, tmp_path: pathlib.Path):
     # Wait for end of shard transfer
     wait_for_collection_shard_transfers_count(peer_api_uris[0], COLLECTION_NAME, 0)
 
-    # Confirm that we fall back to a different method after WAL delta fails
-    stdout, _stderr = capfd.readouterr()
-    assert "Failed to do shard diff transfer, falling back to default method" in stdout
+    # Assume that the WAL delta could not be resolved, preventing a diff
+    # transfer. 'Failed to do shard diff transfer, falling back to default
+    # method' is reported in the logs. But we cannot assert that at this point.
+    # It falls back to streaming records.
+    # TODO(1.9): assert 'streaming_records' transfer method in cluster state
```

Comment:
I'm a bit lost here, why should we only assert this in 1.9? Why can't we do so now?

Reply:
Because in our current implementation we don't properly fall back yet. And the transfers status doesn't show it yet either.

Currently, we simply call the stream records logic once WAL delta transfer fails. We don't change the transfer method through consensus.

In 1.9 we'll change the transfer method through consensus with the new call we implemented.

Reply:
Thanks for clarifying!